### PR TITLE
Updated annoy c library, implemented bindings and tests for hamming

### PR DESCRIPTION
**Pull request by nemo83**

### CODE REVIEW

**Comment by nemo83** [comment]
Hi @pishen , I'd like you to, please, review the code and update to the project.

If happy can you +1? 

I've never done release of the project. do I need anything?

**Review by kevoriordan — APPROVED** [review]

**Comment by pishen** [comment]
Cool. Will check it later.

To release the library with macOS and Linux binaries included, I have to clone and build the code on both macOS and Ubuntu machines with g++, put the binaries together, and release it from my laptop. A little bit cumbersome now. Not sure if there's a better way.

**Comment by nemo83** [comment]
No need to release lib with native implementation. We use Docker and our images will provide the library. I believe the scala lib should be release as is.

Thanks!

File: src/test/scala/annoy4s/AnnoySpec.scala [hunk 1]
```diff
@@ -224,4 +224,45 @@ class AnnoySpec extends FlatSpec with Matchers {
...
+    }
+  }
+
+  it should "create/load and query Hamming file index" in {
+
+    val inputFile = getHammingInputFile
+
+    val outputDir = File.newTemporaryDirectory()
+
+    val annoy = Annoy.create[String](inputFile.pathAsString, 10, outputDir.pathAsString, Manhattan)
```
Comment (pishen) [inline src/test/scala/annoy4s/AnnoySpec.scala:255]: `Manhattan` should be `Hamming`?

**Review by pishen — CHANGES_REQUESTED** [review]

**Comment by pishen** [comment]
Seems that Annoy implemented a `HammingWrapper` to do some transformation from `float` to `uint64_t`. Do we have to implement the transformation if we want to use the Hamming version? 

https://github.com/spotify/annoy/blob/3055c811a2223b9efbd89da8d295ec0c16648ab9/src/annoymodule.cc#L49
I don't understand what exactly is he doing at here (sorry for my poor C++ skill). May take some time to figure it out (if I have time :P)

File: src/test/scala/annoy4s/AnnoySpec.scala (hunk 1 above)
Comment (pishen) [reply src/test/scala/annoy4s/AnnoySpec.scala:255]: Maybe we should design a test case that will not pass `Manhattan` here? Is it possible?

Comment (nemo83) [reply src/test/scala/annoy4s/AnnoySpec.scala:255]: yeah good catch.

**Comment by nemo83** [comment]
@pishen it seems that integrating hamming requires some extra work. I'm gonna roll back the `hamming integration` and only update the annoy code as it should be much faster.

**Comment by pishen** [comment]
@nemo83 Moving the code from `annoymodule.cc` to `annoylib.h` seems to make it work (not sure). I will push the change to another branch for you to review.

**Pull request merged**